### PR TITLE
many: implement alias command

### PR DESCRIPTION
**Pull request by pedronis**
This uses snapstate.Alias over a new /v2/aliases to implement a "snap alias" command. Comes also with spread tests.

### CODE REVIEW

**Comment by pedronis** [comment]
based on #2423

File: tests/main/alias/task.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    . $TESTSLIB/snaps.sh
+    install_local aliases
+
+execute: |
+    echo "Sanity check"
+    aliases.cmd1|MATCH "ok command 1"
+    aliases.cmd2|MATCH "ok command 2"
+
+    echo "Enable aliases"
+    snap alias aliases alias1 alias2
```
Comment (niemeyer) [inline tests/main/alias/task.yaml]: \o/

File: daemon/api.go [hunk 2]
```diff
@@ -2183,3 +2190,48 @@ func getUsers(c *Command, r *http.Request, user *auth.UserState) Response {
 	}
 	return SyncResponse(resp, nil)
 }
+
+// aliasAction is an action performed on aliases
+type aliasAction struct {
+	Action  string   `json:"action"`
+	Snap    string   `json:"snap"`
+	Aliases []string `json:"aliases"`
+}
```
Comment (niemeyer) [inline daemon/api.go]: +1

File: daemon/api_test.go [hunk 3]
```diff
@@ -4620,3 +4620,54 @@ func (s *postCreateUserSuite) TestSysinfoIsManaged(c *check.C) {
...
+	st.Lock()
+	err = chg.Err()
+	st.Unlock()
+	c.Assert(err, check.IsNil)
+
+	// sanity check
+	c.Check(osutil.IsSymlink(filepath.Join(dirs.SnapBinariesDir, "alias1")), check.Equals, true)
+}
+
+func (s *apiSuite) TestAliasErrors(c *check.C) {
```
Comment (niemeyer) [inline daemon/api_test.go]: Anything in mind for here, or should we drop it?

**Pull request merged**